lang/java/reef-utils/src/test/java/org/apache/reef/util/MultiAsyncToSyncTest.java

```
@@ -0,0 +1,225 @@
+   */
+  SynchronousApi(final int incrementerSleepTimeSeconds, final long timeoutPeriodSeconds) {
+    this.incrementerSleepTimeMillis = 1000 * incrementerSleepTimeSeconds;
+    this.blocker = new MultiAsyncToSync(timeoutPeriodSeconds, SECONDS);
+    this.executor = Executors.newFixedThreadPool(2);
+  }
+
+  /**
+   * Asynchronously increment the input parameter.
+   * @param input An integer object whose value is to be incremented by one.
+   * @return The input parameter incremented by one or zero for a timeout.
+   * @throws InterruptedException Thread was interrupted by another thread.
+   * @throws ExecutionException An exception was thrown an internal processing function.
+   */
+  public int apiCall(final Integer input) throws InterruptedException, ExecutionException {
+    // Create a future to run the asynchronous processing.
+    final long identifier = idCounter.getAndIncrement();
+    final FutureTask<Integer> task =
+        new FutureTask<>(new AsynchronousIncrementer(input, identifier, incrementerSleepTimeMillis, blocker));
+    taskQueue.add(task);
+
+    LOG.log(Level.INFO, "Running the incrementer...");
+    // Start the task and block the caller until it completes.
+    executor.execute(task);
+    if (blocker.block(identifier)) {
+      LOG.log(Level.INFO, "Call timed out...");
+      // Timeout occurred before the asynchronous processing completed.
+      return 0;
+    }
+    int result = -1;
+    LOG.log(Level.INFO, "Call getting task result...");
+    result = task.get();
+    return result;
+  }
+
+  /**
+   * Insure all test tasks have completed.
+   */
+  public void complete() throws ExecutionException {
+    try {
+      for (final FutureTask<Integer> task : taskQueue) {
+        task.get();
+      }
+    } catch (ExecutionException ee) {
+      if (!(ee.getCause() instanceof InvalidBlockedCallerIdentifierException)) {
+        throw ee;
+      } else {
+        LOG.log(Level.INFO, "Caught exception waiting for completion...", ee);
```

Please add a comment as to why it is safe to swallow the exception.